project-base/storefront/components/Basic/Flag/Flag.tsx

```diff
@@ -12,15 +12,15 @@ type FlagProps = NativeProps & {
 const getDataTestId = (dataTestId?: string) => dataTestId ?? 'basic-flag';
 
 export const Flag: FC<FlagProps> = ({ children, dataTestId, href, classNames }) => (
-    <ExtendedNextLink type="blogCategory" href={href} passHref>
-        <a
-            className={twMergeCustom(
-                'mb-2 mr-3 inline-flex rounded-sm bg-primaryLight py-1 px-2 text-xs uppercase text-dark no-underline hover:text-dark hover:no-underline',
-                classNames,
-            )}
-            data-testid={getDataTestId(dataTestId)}
-        >
-            {children}
-        </a>
+    <ExtendedNextLink
+        type="blogCategory"
+        href={href}
+        className={twMergeCustom(
+            'mb-2 mr-3 inline-flex rounded-sm bg-primaryLight py-1 px-2 text-xs uppercase text-dark no-underline hover:text-dark hover:no-underline',
+            classNames,
+        )}
+        data-testid={getDataTestId(dataTestId)}
```

Comment:
I changed it to `dataTestId={getDataTestId(dataTestId)}`

Reply:
I am not sure, if you've checked the DOM for this attribute, but I think this will not be enough, or is it? I think you will need to implement it as prop also in `ExtendedNextLink`, like 

```
<NextLink
            as={isStatic ? as : href}
            prefetch={false}
            href={
                isStatic
                    ? href
                    : {
                          pathname: FriendlyPagesDestinations[type],
                          query: { [SLUG_TYPE_QUERY_PARAMETER_NAME]: FriendlyPagesTypes[type], ...queryParams },
                      }
            }
            data-testid={dataTestId}
            {...props}
        >
```

Otherwise it will be just passed to `NextLink` component without being rendered to the DOM as element attribute. But please check it first.

Reply:
I actually reverted it to as it was before, meaning
`data-testid={getDataTestId(dataTestId)}`

This way it is included in the DOM.